spark/src/test/java/org/apache/iceberg/spark/source/TestSparkDataWrite.java

```diff
@@ -0,0 +1,457 @@
+
+    df.select("id", "data", "info").write()
+        .format("iceberg")
+        .option("write-format", format.toString())
+        .mode("append")
+        .save(location.toString());
+
+    table.refresh();
+
+    Dataset<Row> result = spark.read()
+        .format("iceberg")
+        .load(location.toString());
+
+    List<NestedRecord> actual = result.orderBy("id").as(Encoders.bean(NestedRecord.class)).collectAsList();
+    Assert.assertEquals("Number of rows should match", expected.size(), actual.size());
+    Assert.assertEquals("Result rows should match", expected, actual);
+
+    List<DataFile> files = Lists.newArrayList();
+    for (ManifestFile manifest : table.currentSnapshot().manifests()) {
+      for (DataFile file : ManifestReader.read(manifest, table.io())) {
+        files.add(file);
+      }
+    }
+    // TODO: ORC file now not support target file size
+    if (format.equals(FileFormat.PARQUET)) {
+      Assert.assertEquals("Should have 4 DataFiles", 4, files.size());
+      Assert.assertTrue("All DataFiles contain 1000 rows", files.stream().allMatch(d -> d.recordCount() == 1000));
+    }
+  }
+
+  @Test
+  public void testPartitionedCreateWithTargetFileSizeViaOption() throws IOException {
+    File parent = temp.newFolder(format.toString());
+    File location = new File(parent, "test");
+
+    HadoopTables tables = new HadoopTables(CONF);
+    PartitionSpec spec = PartitionSpec.builderFor(SCHEMA).identity("data").build();
+    Table table = tables.create(SCHEMA, spec, location.toString());
+
+    List<NestedRecord> expected = Lists.newArrayListWithCapacity(8000);
+    for (int i = 0; i < 2000; i++) {
+      expected.add(new NestedRecord(i, "a", info));
+      expected.add(new NestedRecord(i, "b", info));
+      expected.add(new NestedRecord(i, "c", info));
+      expected.add(new NestedRecord(i, "d", info));
+    }
+
+    Dataset<Row> df = spark.createDataFrame(expected, NestedRecord.class);
+
+    df.select("id", "data", "info").sort("data").write()
+        .format("iceberg")
+        .option("write-format", format.toString())
+        .mode("append")
+        .option("target-file-size-bytes", 4) // ~4 bytes; low enough to trigger
+        .save(location.toString());
+
+    table.refresh();
+
+    Dataset<Row> result = spark.read()
+        .format("iceberg")
+        .load(location.toString());
+
+    List<NestedRecord> actual = result.orderBy("id").as(Encoders.bean(NestedRecord.class)).collectAsList();
+    Assert.assertEquals("Number of rows should match", expected.size(), actual.size());
+    Assert.assertEquals("Result rows should match", expected, actual);
+
+    List<DataFile> files = Lists.newArrayList();
+    for (ManifestFile manifest : table.currentSnapshot().manifests()) {
+      for (DataFile file : ManifestReader.read(manifest, table.io())) {
+        files.add(file);
+      }
+    }
+    // TODO: ORC file now not support target file size
+    if (format.equals(FileFormat.PARQUET)) {
```

Comment:
Does this not work for Avro as well?

Reply:
Yes now `target-file-size-byte` not work on Avro or Orc file.